### PR TITLE
contrib/syncthing: new package (1.27.1)

### PR DESCRIPTION
**Pull request by mhmdanas**

### CODE REVIEW

File: contrib/syncthing/template.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+url = "https://syncthing.net"
+source = f"https://github.com/syncthing/{pkgname}/archive/v{pkgver}.tar.gz"
+sha256 = "e91671a8291a6d83264f1d1ed731fb6041b25d7259400e5f71cb2d241a48e6a4"
+
+
+def do_build(self):
+    self.do("go", "generate", "github.com/syncthing/syncthing/lib/api/auto")
+    self.golang.build(wrksrc="cmd/stdiscosrv")
+    self.golang.build(wrksrc="cmd/strelaysrv")
+    self.golang.build(wrksrc="cmd/syncthing")
```
Comment (nekopsykose) [inline contrib/syncthing/template.py:28]: you should be able to put these in `make_build_args = ["./cmd/` and do a pre_build for `go generate`

File: contrib/syncthing/template.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+def post_install(self):
+    self.install_license("cmd/strelaysrv/LICENSE", pkgname="syncthing-relaysrv")
+    self.install_file("etc/firewall-ufw/syncthing", "etc/ufw/applications.d")
+    self.install_file(
+        "etc/linux-desktop/*.desktop", "usr/share/applications", glob=True
+    )
+    self.install_file("etc/linux-sysctl/30-syncthing.conf", "usr/lib/sysctl.d")
+
+    self.install_man("man/*.1", glob=True)
```
Comment (nekopsykose) [inline contrib/syncthing/template.py:39]: `*.[157]` lets you do it in 1 install

File: contrib/syncthing/template.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    self.install_file(
+        "assets/logo-only.svg",
+        "usr/share/icons/hicolor/scalable/apps",
+        name="syncthing.svg",
+    )
+
+
+@subpackage("syncthing-relaysrv")
+def _relaysrv(self):
+    self.pkgdesc = f"{pkgdesc} - relay server"
```
Comment (q66) [inline contrib/syncthing/template.py:56]: `{pkgdesc} (relay server)` , ditto for the other one

File: contrib/syncthing/template.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    self.license = "MIT"
```
Comment (q66) [inline contrib/syncthing/template.py:57]: subpackages shouldn't change license

File: contrib/syncthing/template.py [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    "./lib/...",
+    "./cmd/stdiscosrv",
+    "./cmd/strelaysrv",
+    "./cmd/syncthing",
+]
+hostmakedepends = ["go"]
+go_build_tags = ["noupgrade"]
+pkgdesc = "Continuous file synchronization program"
+maintainer = "triallax <triallax@tutanota.com>"
+license = "MPL-2.0"
```
Comment (q66) [inline contrib/syncthing/template.py:21]: MPL-2.0 AND MIT

File: contrib/syncthing/template.py [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+            name="syncthing.png",
+        )
+    self.install_file(
+        "assets/logo-only.svg",
+        "usr/share/icons/hicolor/scalable/apps",
+        name="syncthing.svg",
+    )
+
+
+@subpackage("syncthing-relaysrv")
```
Comment (q66) [inline contrib/syncthing/template.py:54]: is there a good reason to split these?

File: contrib/syncthing/template.py (hunk 4 above)
Comment (q66) [reply contrib/syncthing/template.py:57]: actually i guess it's fine for this one, let's leave it

File: contrib/syncthing/template.py (hunk 6 above)
Comment (q66) [reply contrib/syncthing/template.py:54]: ah, they are all massive, nevermind

**Comment by q66** [comment]
lgdm, just fix the pkgdescs

**Pull request merged**